### PR TITLE
[ENHANCEMENT] added support for databricks dialect if using databricks-dbapi sqlalchemy

### PR DESCRIPTION
**Pull request by utkarshshukla2912**
Changes proposed in this pull request:
- While using custom sql for databricks through databricks-dbapi sqlalchemy, a temporary table is created which leads to an error since temporary table is not supported by spark, context around it [here](https://issues.apache.org/jira/browse/SPARK-14651), but temporary view is. Hence have added dialect support for databricks which works on temporary views rather than temporary tables.

- Added the below code since when using databricks `self.sql_engine_dialect.name.lower()` return a byte string.
```
engine_dialect = self.sql_engine_dialect.name.lower()
        if type(engine_dialect) != str:
            engine_dialect = str(engine_dialect, "utf-8")
```

### CODE REVIEW

**Comment by eugmandel** [comment]
@utkarshshukla2912 The PR is in review and @roblim is handling it.

File: great_expectations/dataset/sqlalchemy_dataset.py [hunk 1]
```diff
@@ -1242,11 +1242,19 @@ def create_temporary_table(self, table_name, custom_sql, schema_name=None):
         # similar, for example.
         ###
 
-        if self.sql_engine_dialect.name.lower() == "bigquery":
+        engine_dialect = self.sql_engine_dialect.name.lower()
+        if type(engine_dialect) != str:
+            engine_dialect = str(engine_dialect, "utf-8")
+
+        if engine_dialect == "bigquery":
```
Comment (roblim) [inline great_expectations/dataset/sqlalchemy_dataset.py]:
```suggestion
        engine_dialect = self.sql_engine_dialect.name.lower()
        # handle cases where dialect.name.lower() returns a byte string (e.g. databricks)
        if isinstance(engine_dialect, bytes):
            engine_dialect = str(engine_dialect, "utf-8")

        if engine_dialect == "bigquery":
```

Stylistically, we prefer using `isinstance` to check types, vs. the `type(x) == y` or `type(x) != y` pattern.

Comment (roblim) [reply great_expectations/dataset/sqlalchemy_dataset.py]: Also suggest adding comment for clarity.

**Review by roblim — CHANGES_REQUESTED** [review]
Looks good to me. Just had one stylistic suggestion. Thanks so much for your contribution!

**Comment by utkarshshukla2912** [comment]
@roblim have added your suggestions, and updated with the latest develop branch.

**Review by roblim — APPROVED** [review]

**Pull request merged**